source/common/upstream/cluster_manager_impl.cc

```diff
@@ -211,7 +211,7 @@ ClusterManagerImpl::ClusterManagerImpl(const envoy::config::bootstrap::v2::Boots
   // Now setup ADS if needed, this might rely on a primary cluster.
   if (bootstrap.dynamic_resources().has_ads_config()) {
     ads_mux_.reset(new Config::GrpcMuxImpl(
-        bootstrap.node(),
+        local_info,
```

Comment:
I have mixed opinions on this change. On the one hand, we make it more full proof, by forcing the `LocalInfo` type. On the other, we are now passing a strict superset of what we were passing before, and the new information is not used. My main thoughts on making `LocalInfo` canonical would be to just delete the node info from bootstrap very early in server init. But, this approach also works..

Reply:
Interesting POV, will try to explore that too.

Reply:
It's not necessarily the correct PoV; just sharing where my thinking was originally :)